apollo/lib/router.py

```diff
@@ -79,3 +76,18 @@ def wrapped(request: Request, *args, **kwargs):
 
         wrapped.__signature__ = new_signature
         return wrapped
+
+    def _get_wrapped(self, func, route, permission,
```

Comment:
btw I just needed to decorate an async method too. I'd just:
```
    def _http_method(self, func, http_method, *outer_args, permission='public',
                     **outer_kwargs):
        route = getattr(super(SecureRouter, self), http_method)

        @route(*outer_args, **outer_kwargs)
        async def wrapped(request: Request, *args, **kwargs):
            self.acl_policy.validate_permission(permission, request)
            output = func(*args, **kwargs)
            if inspect.iscoroutine(output):
                output = await output

            return output

        ...
```

Saves you a lot of test duplication too because the wrapper is always async.

Reply:
Ait thankss